### ISSUE TITLE
Inconsistent handling of constant_score.filter

### ISSUE THREAD

**Issue by roytmana** [elastic/elasticsearch#19543]
**Elasticsearch version**:2.3.4

**JVM version**:1.7 latest

**OS version**:Win10

**Description of the problem including expected versus actual behavior**:
While bool query "filter" accepts array of queries and use AND to combine them,
constant_score and filtered query "filter" element also takes array of queries without exception but result seem to be non deterministic - neither "OR" no "AND" not even sure what.

Example - following query should return 0 hits but it returns number of hits one of the queries would produce and not even consistently first or second

```
{
  "query": {
    "constant_score": {
      "filter": [
        {
          "terms": {
            "date.fy": [
              2016
            ]
          }
        },
        {
          "terms": {
            "date.fy": [
              2015
            ]
          }
        }
      ]
    }
  }
}
```

It would be very nice to have it support array of queries like bool.filter does or at least thro exception if it is considered invalid

Also in case of bool.filter (and bool.must), use of array is not clearly documented https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html

**Comment by jpountz** [comment]
This is a bug indeed, `constant_score` should reject arrays.

**Comment by roytmana** [comment]
Filtered query has the same issue.
I wonder if making it treat array same as boolean.filter would be less breaking than growing exception

**Comment by clintongormley** [comment]
Note: filters are deprecated and have been removed in 5.0.  Also, this bug has been fixed in 5.0

**Comment by colings86** [comment]
Closing since this is fixed in 5.0+
